### PR TITLE
DEVPROD-2417 Add final block to task details

### PR DESCRIPTION
**Pull request by ZackarySantana**
DEVPROD-2417

### Description
To enable some use-cases where automation could rely on what block a task failed on, this propogates the block scope through the task context to send to the evergreen app server.

### Testing
Unit testing and manual testing on staging, here is an example json response (see `final_block_type`)

EDIT: View comments below, this was changed to `error_block_type`. The following json is only displayed for history:
```json
{
  "task_id": "evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17",
  "project_id": "mci",
  "project_identifier": "evg",
  "create_time": "2024-02-29T15:44:40Z",
  "dispatch_time": "2024-03-22T18:58:02.976Z",
  "scheduled_time": "2024-03-22T18:56:24.535Z",
  "container_allocated_time": null,
  "start_time": "2024-03-22T18:58:03.645Z",
  "finish_time": "2024-03-22T18:58:03.989Z",
  "ingest_time": "2024-03-22T18:56:17.518Z",
  "activated_time": "2024-03-22T18:56:17.518Z",
  "version_id": "65fdd44ab7233f0007285364",
  "revision": "6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685",
  "priority": 0,
  "activated": true,
  "stepback_info": {
    "last_failing_stepback_task_id": "",
    "last_passing_stepback_task_id": "",
    "next_stepback_task_id": "",
    "previous_stepback_task_id": ""
  },
  "activated_by": "zackary.santana",
  "container_allocated": false,
  "container_allocation_attempts": 0,
  "build_id": "evg_ubuntu2204_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17",
  "distro_id": "ubuntu2204-large",
  "container": "",
  "container_options": {
    "cpu": 0,
    "memory_mb": 0,
    "working_dir": "",
    "image": "",
    "os": "",
    "arch": "",
    "windows_version": ""
  },
  "build_variant": "ubuntu2204",
  "build_variant_display_name": "ubuntu2204",
  "depends_on": null,
  "display_name": "my-task",
  "host_id": "i-0071adc871b24c3d7",
  "pod_id": "",
  "execution": 0,
  "order": 138,
  "status": "failed",
  "display_status": "setup-failed",
  "status_details": {
    "status": "failed",
    "type": "setup",
    "desc": "this should be set",
    "final_block_type": "pre",
    "timed_out": false,
    "timeout_type": "",
    "oom_tracker_info": {
      "detected": false,
      "pids": null
    },
    "trace_id": "11896f273b77d33924113cd2fe1d9adf",
    "disk_devices": null
  },
  "logs": {
    "all_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/0?type=ALL",
    "task_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/0?type=T",
    "agent_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/0?type=E",
    "system_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/0?type=S",
    "event_log": "https://evergreen-staging.corp.mongodb.com/event_log/task/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17"
  },
  "parsley_logs": {
    "all_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/0/all",
    "task_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/0/task",
    "agent_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/0/agent",
    "system_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/0/system"
  },
  "time_taken_ms": 344,
  "expected_duration_ms": 10188,
  "est_wait_to_start_ms": 0,
  "generate_task": false,
  "generated_by": "",
  "artifacts": null,
  "display_only": false,
  "parent_task_id": "",
  "mainline": false,
  "blocked": false,
  "requester": "patch_request",
  "test_results": null,
  "aborted": false,
  "abort_info": {},
  "sync_at_end_opts": {
    "enabled": false,
    "statuses": null,
    "timeout": 0
  },
  "ami": "ami-0a9f252ef23bced59",
  "must_have_test_results": false,
  "base_task": {
    "id": null,
    "status": null
  },
  "reset_when_finished": false,
  "archived": false
}
```

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -440,6 +440,7 @@ func (a *Agent) setupTask(agentCtx, setupCtx context.Context, initialTC *taskCon
 	}
 	if factory != nil {
 		tc.setCurrentCommand(factory())
+		tc.setCurrentBlock("setup.initial")
```
Comment (hadjri) [inline agent/agent.go:443]: setup.initial is a command not a block, so I would use "pre" here

**Review by hadjri — CHANGES_REQUESTED** [review]
Some thoughts about the approach: as part of the ask of this ticket, we're not concerned with knowing the last block that a task _ran_, but just whether the task encountered a failure in the post block. Adding a new field to the end task response as you did makes sense, but perhaps for simplicity we could just make it a boolean that indicates if the post block saw an error

Additionally, can we add documentation to the GET task rest route for this field and how to interpret it? And can we confirm in staging that this works specifically for a task that sends a success status but has a failure in post?

**Review by hadjri — CHANGES_REQUESTED** [review]
Some thoughts about the approach: as part of the ask of this ticket, we're not concerned with knowing the last block that a task ran, but just whether the task encountered a failure in the post block. Adding a new field to the end task response as you did makes sense, but perhaps for simplicity we could just make it a boolean that indicates if the post block saw an error

Additionally, can we add documentation to the GET task rest route for this field and how to interpret it? And can we confirm in staging that this works specifically for a task that sends a success status but has a failure in post?

**Comment by ZackarySantana** [comment]
> Some thoughts about the approach: as part of the ask of this ticket, we're not concerned with knowing the last block that a task ran, but just whether the task encountered a failure in the post block. Adding a new field to the end task response as you did makes sense, but perhaps for simplicity we could just make it a boolean that indicates if the post block saw an error

I agree that having a `post_command_failed` / `post_block_failed` field with a true/false value is simpler for Geoff's use case but I think just exposing the block helps keep our codebase simpler. I think it's easier to build upon and is more useful information- for example we could include in the UI for failures what block type it is now. What do you think? I'm not attached to leaving it as the final block type so I can switch it over but I'm not sure it's the right move to.

File: agent/agent.go [hunk 2]
```diff
@@ -440,6 +440,7 @@ func (a *Agent) setupTask(agentCtx, setupCtx context.Context, initialTC *taskCon
 	}
```
Comment (ZackarySantana) [inline agent/agent.go:440]: I'm leaving this as a comment so it is collapsable (as with the others)

This is a task that succeeded (and it had a post command):

```
{
  "task_id": "evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01",
  "project_id": "mci",
  "project_identifier": "evg",
  "create_time": "2024-02-29T15:44:40Z",
  "dispatch_time": "2024-03-28T22:56:37.301Z",
  "scheduled_time": "2024-03-28T22:55:54.88Z",
  "container_allocated_time": null,
  "start_time": "2024-03-28T22:56:37.69Z",
  "finish_time": "2024-03-28T22:56:40.768Z",
  "ingest_time": "2024-03-28T22:21:01.818Z",
  "activated_time": "2024-03-28T22:55:52.656Z",
  "version_id": "6605ed477e708300076689cd",
  "revision": "6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685",
  "priority": 0,
  "activated": true,
  "stepback_info": {
    "last_failing_stepback_task_id": "",
    "last_passing_stepback_task_id": "",
    "next_stepback_task_id": "",
    "previous_stepback_task_id": ""
  },
  "activated_by": "zackary.santana",
  "container_allocated": false,
  "container_allocation_attempts": 0,
  "build_id": "evg_ubuntu2204_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01",
  "distro_id": "ubuntu2204-small",
  "container": "",
  "container_options": {
    "cpu": 0,
    "memory_mb": 0,
    "working_dir": "",
    "image": "",
    "os": "",
    "arch": "",
    "windows_version": ""
  },
  "build_variant": "ubuntu2204",
  "build_variant_display_name": "ubuntu2204",
  "depends_on": null,
  "display_name": "my-task",
  "host_id": "i-0b1a77c0db5166034",
  "pod_id": "",
  "execution": 1,
  "order": 142,
  "status": "success",
  "display_status": "success",
  "status_details": {
    "status": "success",
    "type": "",
    "desc": "",
    "error_block_type": "",
    "timed_out": false,
    "timeout_type": "",
    "oom_tracker_info": {
      "detected": false,
      "pids": null
    },
    "trace_id": "ad8a29ede4d6667b7cb56ed823b71b41",
    "disk_devices": [
      "xvda1",
      "xvdb"
    ]
  },
  "logs": {
    "all_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1?type=ALL",
    "task_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1?type=T",
    "agent_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1?type=E",
    "system_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1?type=S",
    "event_log": "https://evergreen-staging.corp.mongodb.com/event_log/task/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01"
  },
  "parsley_logs": {
    "all_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1/all",
    "task_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1/task",
    "agent_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1/agent",
    "system_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed477e708300076689cd_24_03_28_22_21_01/1/system"
  },
  "time_taken_ms": 3078,
  "expected_duration_ms": 12927,
  "est_wait_to_start_ms": 0,
  "generate_task": false,
  "generated_by": "",
  "artifacts": null,
  "display_only": false,
  "parent_task_id": "",
  "mainline": false,
  "blocked": false,
  "requester": "patch_request",
  "test_results": null,
  "aborted": false,
  "abort_info": {},
  "sync_at_end_opts": {
    "enabled": false,
    "statuses": null,
    "timeout": 0
  },
  "ami": "ami-0a9f252ef23bced59",
  "must_have_test_results": false,
  "base_task": {
    "id": null,
    "status": null
  },
  "reset_when_finished": false,
  "archived": false
}
```

Comment (ZackarySantana) [reply agent/agent.go:440]: A task that failed in the post: 

```
{
  "task_id": "evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45",
  "project_id": "mci",
  "project_identifier": "evg",
  "create_time": "2024-02-29T15:44:40Z",
  "dispatch_time": "2024-03-28T22:56:40.939Z",
  "scheduled_time": "2024-03-28T22:55:54.88Z",
  "container_allocated_time": null,
  "start_time": "2024-03-28T22:56:41.569Z",
  "finish_time": "2024-03-28T22:56:44.887Z",
  "ingest_time": "2024-03-28T22:21:45.025Z",
  "activated_time": "2024-03-28T22:55:50.016Z",
  "version_id": "6605ed727e708300076689d4",
  "revision": "6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685",
  "priority": 0,
  "activated": true,
  "stepback_info": {
    "last_failing_stepback_task_id": "",
    "last_passing_stepback_task_id": "",
    "next_stepback_task_id": "",
    "previous_stepback_task_id": ""
  },
  "activated_by": "zackary.santana",
  "container_allocated": false,
  "container_allocation_attempts": 0,
  "build_id": "evg_ubuntu2204_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45",
  "distro_id": "ubuntu2204-small",
  "container": "",
  "container_options": {
    "cpu": 0,
    "memory_mb": 0,
    "working_dir": "",
    "image": "",
    "os": "",
    "arch": "",
    "windows_version": ""
  },
  "build_variant": "ubuntu2204",
  "build_variant_display_name": "ubuntu2204",
  "depends_on": null,
  "display_name": "my-task",
  "host_id": "i-0b1a77c0db5166034",
  "pod_id": "",
  "execution": 2,
  "order": 143,
  "status": "failed",
  "display_status": "failed",
  "status_details": {
    "status": "failed",
    "type": "test",
    "desc": "'shell.exec' (step 1 of 1) in block 'post'",
    "error_block_type": "post",
    "timed_out": false,
    "timeout_type": "",
    "oom_tracker_info": {
      "detected": false,
      "pids": null
    },
    "trace_id": "ecbfde5ad54ea642e08c1a878e61ce0d",
    "disk_devices": [
      "xvda1",
      "xvdb"
    ]
  },
  "logs": {
    "all_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2?type=ALL",
    "task_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2?type=T",
    "agent_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2?type=E",
    "system_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2?type=S",
    "event_log": "https://evergreen-staging.corp.mongodb.com/event_log/task/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45"
  },
  "parsley_logs": {
    "all_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2/all",
    "task_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2/task",
    "agent_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2/agent",
    "system_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_6605ed727e708300076689d4_24_03_28_22_21_45/2/system"
  },
  "time_taken_ms": 3318,
  "expected_duration_ms": 12927,
  "est_wait_to_start_ms": 0,
  "generate_task": false,
  "generated_by": "",
  "artifacts": null,
  "display_only": false,
  "parent_task_id": "",
  "mainline": false,
  "blocked": false,
  "requester": "patch_request",
  "test_results": null,
  "aborted": false,
  "abort_info": {},
  "sync_at_end_opts": {
    "enabled": false,
    "statuses": null,
    "timeout": 0
  },
  "ami": "ami-0a9f252ef23bced59",
  "must_have_test_results": false,
  "base_task": {
    "id": null,
    "status": null
  },
  "reset_when_finished": false,
  "archived": false
}
```

Comment (ZackarySantana) [reply agent/agent.go:440]: A task that failed in the pre
```
{
  "task_id": "evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17",
  "project_id": "mci",
  "project_identifier": "evg",
  "create_time": "2024-02-29T15:44:40Z",
  "dispatch_time": "2024-03-28T22:56:45.066Z",
  "scheduled_time": "2024-03-28T22:55:50.652Z",
  "container_allocated_time": null,
  "start_time": "2024-03-28T22:56:45.604Z",
  "finish_time": "2024-03-28T22:56:45.924Z",
  "ingest_time": "2024-03-22T18:56:17.518Z",
  "activated_time": "2024-03-28T22:55:48.225Z",
  "version_id": "65fdd44ab7233f0007285364",
  "revision": "6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685",
  "priority": 0,
  "activated": true,
  "stepback_info": {
    "last_failing_stepback_task_id": "",
    "last_passing_stepback_task_id": "",
    "next_stepback_task_id": "",
    "previous_stepback_task_id": ""
  },
  "activated_by": "zackary.santana",
  "container_allocated": false,
  "container_allocation_attempts": 0,
  "build_id": "evg_ubuntu2204_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17",
  "distro_id": "ubuntu2204-small",
  "container": "",
  "container_options": {
    "cpu": 0,
    "memory_mb": 0,
    "working_dir": "",
    "image": "",
    "os": "",
    "arch": "",
    "windows_version": ""
  },
  "build_variant": "ubuntu2204",
  "build_variant_display_name": "ubuntu2204",
  "depends_on": null,
  "display_name": "my-task",
  "host_id": "i-0b1a77c0db5166034",
  "pod_id": "",
  "execution": 2,
  "order": 138,
  "status": "failed",
  "display_status": "setup-failed",
  "status_details": {
    "status": "failed",
    "type": "setup",
    "desc": "this should be set",
    "error_block_type": "pre",
    "timed_out": false,
    "timeout_type": "",
    "oom_tracker_info": {
      "detected": false,
      "pids": null
    },
    "trace_id": "0cb27240f49a758c80b29e0cc9a354fb",
    "disk_devices": [
      "xvda1",
      "xvdb"
    ]
  },
  "logs": {
    "all_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2?type=ALL",
    "task_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2?type=T",
    "agent_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2?type=E",
    "system_log": "https://evergreen-staging.corp.mongodb.com/task_log_raw/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2?type=S",
    "event_log": "https://evergreen-staging.corp.mongodb.com/event_log/task/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17"
  },
  "parsley_logs": {
    "all_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2/all",
    "task_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2/task",
    "agent_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2/agent",
    "system_log": "https://parsley-staging.corp.mongodb.com/evergreen/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_65fdd44ab7233f0007285364_24_03_22_18_56_17/2/system"
  },
  "time_taken_ms": 320,
  "expected_duration_ms": 12000,
  "est_wait_to_start_ms": 0,
  "generate_task": false,
  "generated_by": "",
  "artifacts": null,
  "display_only": false,
  "parent_task_id": "",
  "mainline": false,
  "blocked": false,
  "requester": "patch_request",
  "test_results": null,
  "aborted": false,
  "abort_info": {},
  "sync_at_end_opts": {
    "enabled": false,
    "statuses": null,
    "timeout": 0
  },
  "ami": "ami-0a9f252ef23bced59",
  "must_have_test_results": false,
  "base_task": {
    "id": null,
    "status": null
  },
  "reset_when_finished": false,
  "archived": false
}
```

File: agent/agent.go [hunk 3]
```diff
@@ -973,14 +975,16 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string,
 		if err := a.runPostOrTeardownTaskCommands(ctx, tc); err != nil {
 			tc.logger.Task().Info("Post task completed - FAILURE. Overall task status changed to FAILED.")
 			setEndTaskFailureDetails(tc, detail, evergreen.TaskFailed, "", "")
+			detail.ErrorBlockType = string(tc.getCurrentBlock())
```
Comment (hadjri) [inline agent/agent.go:978]: I suppose I'm fine with this approach, but what this approach needs to accomplish is it needs to be able to indicate whether post commands failed even when the task succeeded (i.e., even if `post_error_fails_task` = false). The casing [here](https://gist.github.com/hadjri/8b8b99be1521cd43af9c65a3f82c8a4f#file-agent-go-L859) will need to check to update the failing block type regardless of whether `post.canFailTask`

**Review by hadjri — CHANGES_REQUESTED** [review]

File: agent/agent.go (hunk 3 above)
Comment (ZackarySantana) [reply agent/agent.go:978]: Ah I see, I did not realize that's the point. I think I agree with your earlier comment more now to just simplify it and to update that casting you linked to set it as well!

**Comment by ZackarySantana** [comment]
Re-tested in staging, here's an example task: https://spruce-staging.corp.mongodb.com/task/evg_ubuntu2204_my_task_patch_6107b5deeeb8b6f9d0b0563dcebf8ec7b2085685_660eef8ced22c900072a41cb_24_04_10_16_03_47/logs?execution=0

That through the api, it has post_errored true!

**Review by hadjri — APPROVED** [review]
LGTM - once the change is deployed you can point the user to the updated [API swagger schema](https://mciuploads.s3.amazonaws.com/evergreen/latest/swagger.json) which should include the new PostErrored field and its description.

**Pull request merged**